tools/permanence-calculator/components/controls/slider.js

```diff
@@ -43,6 +45,43 @@ const Control = ({
     }
   }, [active])
 
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    updateParamValueFromInput()
+  }
+
+  const updateParamValueFromInput = () => {
+    let v = parseFloat(displayValue)
+    if (!isNaN(v)) {
+      if (v < min) {
+        v = min
+      }
+      if (v > max) {
+        v = max
+      }
+      setValue(v)
+      setDisplayValue(v)
+    } else {
+      setDisplayValue(value)
+    }
+  }
+
+  const updateParamValue = (e) => {
+    setValue(parseFloat(e.target.value))
+  }
+
+  const updateParamDisplayValue = (e) => {
+    let normalized = e.target.value
+    if (name === 'discount rate' || name === 'project risk') {
+      normalized = normalized.replace('%', '')
```

Comment:
Hm I'm seeing slightly funny behavior where editing the field feels like a noop if I start by deleting the last character (`%`) -- is this working like you'd it expect it to?

Reply:
Ha, so, yes that was intentional. I actually did it by copying how you handled something similar with the donation page, where deleting the `$` is a noop. It's slightly weirder here because it's at the end instead of the start, but I couldn't figure out another way to then always render with a `%`. Might work to have a state for whether you're editing or not, and skip appending the `%` while editing. On the other hand, it's kinda nice to always see the `%`?

Reply:
Yeah I think the extra character being at the end makes the treatment meaningfully clunkier.
> Might work to have a state for whether you're editing or not, and skip appending the % while editing
This sounds reasonable to me! Put another way, we'd be adding the `%` onBlur if it's not already present?